Assets/Scripts/Manager/GhostSwitchManager.cs

```diff
@@ -126,7 +126,9 @@ public void updateDolls ()
 	// make the camera follow the current player
 	void updateCamera ()
 	{
-		Camera.main.GetComponent<CameraFollow> ().findPlayer ();
+        CameraFollow cameraFollow = Camera.main.GetComponent<CameraFollow>();
+        cameraFollow.findPlayer ();
+        cameraFollow.setCurrentDollSpeed(player.GetComponent<Controller2>().maxSpeedSlowerBy);
 	}
```

Comment:
if you make the changes I suggested, here you can just findPlayer() as you assume that the method take care of changing the speed on its own